Microsoft.Toolkit.Uwp.UI/AdvancedCollectionView/AdvancedCollectionView.cs

```diff
@@ -368,19 +368,34 @@ int IComparer<object>.Compare(object x, object y)
                 var typeInfo = x.GetType().GetTypeInfo();
                 foreach (var sd in _sortDescriptions)
                 {
-                    _sortProperties[sd.PropertyName] = typeInfo.GetDeclaredProperty(sd.PropertyName);
+                    if (!string.IsNullOrEmpty(sd.PropertyName))
+                    {
+                        _sortProperties[sd.PropertyName] = typeInfo.GetDeclaredProperty(sd.PropertyName);
+                    }
                 }
             }
 
             foreach (var sd in _sortDescriptions)
             {
-                var pi = _sortProperties[sd.PropertyName];
-                var cx = pi.GetValue(x) as IComparable;
-                var cy = pi.GetValue(y) as IComparable;
+                object cx, cy;
+
+                if (string.IsNullOrEmpty(sd.PropertyName))
+                {
+                    cx = x;
+                    cy = y;
+                }
+                else
+                {
+                    var pi = _sortProperties[sd.PropertyName];
+
+                    cx = pi.GetValue(x);
+                    cy = pi.GetValue(y);
+                }
+
                 try
```

Comment:
If there's any chance this will make it into the current release, can we remove this try/catch? All it ends up doing is catching user errors that should be surfaced.